### PR TITLE
MNT: Drop travis for GHA

### PR DESCRIPTION
**Pull request by maffettone**
Remove Travis for CI/CD and replace with GHA. Actions for the following were added: 
1. flake: code quality
2. tests: run the same `run_tests.py` script that was run by Travis.
3. python-publish: Publish to PyPI when a release is issued. 

Notably, this shows some lint that wasn't highlighted by Travis (Flake8), and a couple of failing unit tests...

### CODE REVIEW

**Review by mrakitin — COMMENTED** [review]
Looks good overall, thanks for the effort, @maffettone!

Maybe we could limit the version of NumPy in the requirements to avoid test failures? 

What's the most recent version of NumPy where it works? We now have `numpy=1.24.4` in our latest conda envs [2023-3.2](https://zenodo.org/records/10056340). The latest version is https://pypi.org/project/numpy/1.26.1/.

File: .github/workflows/flake8.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        run: |
+          # These packages are installed in the base environment but may be older
+          # versions. Explicitly upgrade them because they often create
+          # installation problems if out of date.
+          python -m pip install --upgrade pip setuptools numpy
+
+          pip install flake8
+      - name: Run flake8
+        run: |
+          flake8
```
Comment (ambarb) [inline .github/workflows/flake8.yml:23]: excellent to add this!  I clearly forgot to do it with the last PR

the helpers modules are the only thing affected (my addition).

so the question is how to resolve the CI error for flake8?  The options are:
1. add it to issues and take care of it when I fix the bug in helpers.overscan
2. fix it now

Option 2, fix it now depends on who can do what and who is willing to wait. 
1. I can checkout this branch fix flake8
2. we (I or someone else) can make a different branch that gets merged now and then Phil can pull in those changes
3. Phil fix in place (but I don't know if he is available to do that on the timeline we need)

@mrakitin are you hoping for this to be done mid-next week (Nov 8)?  Depending on that and who we agree does what, I think we can move forward without make a mess of two people trying to fix the same thing.

**Review by ambarb — CHANGES_REQUESTED** [review]

File: .github/workflows/flake8.yml (hunk 1 above)
Comment (ambarb) [reply .github/workflows/flake8.yml:23]: only can click "resolve conversation".  If I do that now, my comment collapses.  Sorry about that.

Comment (mrakitin) [reply .github/workflows/flake8.yml:23]: I think it's not that time-pressing. We can resolve the linting/styling issues later as the release of v0.2.1 is already on PyPI, so it unblocks our work for raydata.

Comment (ambarb) [reply .github/workflows/flake8.yml:23]: great.  added issue so will resolve here

**Comment by maffettone** [comment]
@mrakitin Looks like the build for the test container has `numpy=1.24.4` so this may take a deeper look. 

I updated the `setup.py` to be aligned with cookie-cutter. Can we merge this, and I'll add some unit testing to my TODOs this week?

File: setup.py [hunk 2]
```diff
@@ -1,45 +1,81 @@
...
+
+pip install --upgrade pip
+""".format(
+        *(sys.version_info[:2] + min_version)
+    )
+    sys.exit(error)
+
+here = path.abspath(path.dirname(__file__))
+
+with open(path.join(here, "README.md"), encoding="utf-8") as readme_file:
```
Comment (mrakitin) [inline setup.py:31]: The markdown format requires an extra kwarg, see https://github.com/NSLS-II/scientific-python-cookiecutter/issues/25.

File: setup.py [hunk 3]
```diff
@@ -1,45 +1,81 @@
...
     version=versioneer.get_version(),
     cmdclass=versioneer.get_cmdclass(),
-    author='Brookhaven National Laboratory',
-    packages=setuptools.find_packages(exclude=['src', 'tests']),
-    ext_package='csxtools.ext',
+    author="Brookhaven National Laboratory",
+    description="Python library for tools to be used at the Coherent Soft X-ray scattering (CSX) beamline at NSLS-II.",
+    packages=setuptools.find_packages(exclude=["src", "tests"]),
+    python_requires=">={}".format(".".join(str(n) for n in min_version)),
+    long_description=readme,
```
Comment (mrakitin) [inline setup.py:66]:
```suggestion
    long_description=readme,
    long_description_content_type='text/markdown',
```

**Review by mrakitin — COMMENTED** [review]
Looks good. Let's fix the minor problem with the content type (suggested below). Then it's good to go.

**Review by mrakitin — APPROVED** [review]

**Review by ambarb — APPROVED** [review]

**Comment by ambarb** [comment]
tests are failing due to deprecation.  we will address the deprecated unit tests separately.
https://github.com/NSLS-II-CSX/csxtools/issues/84
this should be a priority

flake8 is agreed to be addressed later
https://github.com/NSLS-II-CSX/csxtools/issues/93

merging now

**Pull request merged**